src/test/java/seedu/address/model/TutorsPetTest.java

```diff
@@ -82,20 +94,120 @@ public void getStudentList_modifyList_throwsUnsupportedOperationException() {
         assertThrows(UnsupportedOperationException.class, () -> tutorsPet.getStudentList().remove(0));
     }
 
+    //// moduleClass-related tests
+
+    @Test
+    public void resetData_withDuplicateModuleClass_throwsDuplicateModuleClassException() {
+        // Two moduleClasses with the same identity fields
+        ModuleClass editedCS2103T = new ModuleClassBuilder(CS2103T_TUTORIAL).withStudentIds().build();
+        List<Student> defaultStudents = TypicalStudent.getTypicalStudents();
+        List<ModuleClass> newModuleClasses = Arrays.asList(CS2103T_TUTORIAL, editedCS2103T);
+        TutorsPetStub newData = new TutorsPetStub(defaultStudents, newModuleClasses);
+
+        assertThrows(DuplicateModuleClassException.class, () -> tutorsPet.resetData(newData));
+    }
```

Comment:
Would this be a better name?
```suggestion
    public void resetData_withDuplicateModuleClasses_throwsDuplicateModuleClassException() {
        // Two moduleClasses with the same identity fields
        ModuleClass editedCS2103T = new ModuleClassBuilder(CS2103T_TUTORIAL).withStudentIds().build();
        List<Student> defaultStudents = TypicalStudent.getTypicalStudents();
        List<ModuleClass> newModuleClasses = Arrays.asList(CS2103T_TUTORIAL, editedCS2103T);
        TutorsPetStub newData = new TutorsPetStub(defaultStudents, newModuleClasses);

        assertThrows(DuplicateModuleClassException.class, () -> tutorsPet.resetData(newData));
    }
```